os/binfmt/libelf/libelf_bind.c

```diff
@@ -277,6 +281,69 @@ static int elf_relocateadd(FAR struct elf_loadinfo_s *loadinfo, int relidx, FAR
 	return -ENOSYS;
 }
 
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+static int export_library_symtab(FAR struct elf_loadinfo_s *loadinfo)
+{
+	int i;
+	int ret;
+
+	Elf32_Shdr *symtab = &loadinfo->shdr[loadinfo->symtabidx];
+	int nsyms = symtab->sh_size / sizeof(Elf32_Sym);
+
+	g_lib_symtab = kmm_malloc(nsyms * sizeof(struct symtab_s));
+	if (!g_lib_symtab) {
+		berr("Failed to allocate memory for exporting symbol table\n");
+		return -ENOMEM;
+	}
+
+	g_num_lib_syms = 0;
+	for (i = 0; i < nsyms; i++) {
+		Elf32_Sym sym;
+		ret = elf_readsym(loadinfo, i, &sym);
+                if (ret < 0) {
+                        berr("Failed to read symbol[%d]: %d\n", i, ret);
+                        goto ret_err;
+                }
+
+		if (/*ELF32_ST_TYPE(sym.st_info) == STT_FUNC &&*/ ELF32_ST_BIND(sym.st_info) == STB_GLOBAL) {
+			ret = elf_symname(loadinfo, &sym);
+	                if (ret < 0) {
+				berr("SHN_UNDEF: Failed to get symbol name: %d\n", ret);
+				goto ret_err;
+			}
+
+			g_lib_symtab[g_num_lib_syms].sym_name = kmm_zalloc(strlen(loadinfo->iobuffer) + 1);
+			if (!g_lib_symtab[g_num_lib_syms].sym_name) {
+				berr("Error allocating symbol name %d entry\n", g_num_lib_syms);
+				ret = -ENOMEM;
+				goto ret_err;
+			}
+
+			strcpy(g_lib_symtab[g_num_lib_syms].sym_name, loadinfo->iobuffer);
```

Comment:
Please use `strncpy` instead of `strcpy`.

Reply:
We dont know the length of the iobuffer (symbol name). Also, all symbol names in the elf string table are always null terminated. So, it must be fine to perform strcpy. Also, the sym_name variable is allocated based on the `strlen(loadinfo->iobuffer) + 1`.

Reply:
I agree with kishore-sn